oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMakerLargeStringPropertiesLogTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.jackrabbit.oak.plugins.index.lucene;
+
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import org.apache.jackrabbit.oak.api.Type;
+import org.apache.jackrabbit.oak.plugins.index.lucene.util.IndexDefinitionBuilder;
+import org.apache.jackrabbit.oak.spi.state.NodeBuilder;
+import org.apache.jackrabbit.oak.spi.state.NodeState;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.InvalidParameterException;
+
+import static java.util.Arrays.asList;
+import static org.apache.jackrabbit.oak.InitialContentHelper.INITIAL_CONTENT;
+import static org.apache.jackrabbit.oak.plugins.memory.EmptyNodeState.EMPTY_NODE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class LuceneDocumentMakerLargeStringPropertiesLogTest {
+
+    ListAppender<ILoggingEvent> listAppender = null;
+    private final String nodeImplLogger = "org.apache.jackrabbit.oak.plugins.index.lucene.LuceneDocumentMaker";
+    private final String warnMessage = "String length: {} for property: {} at Node: {} is greater than configured value {}";
+    private String customStringPropertyThresholdLimit = "9";
+    private String smallStringProperty = "1234567";
+    private String largeStringPropertyAsPerCustomThreshold = "1234567890";
+
+    @Before
+    public void loggingAppenderStart() {
+        LoggerContext context = (LoggerContext) LoggerFactory.getILoggerFactory();
+        listAppender = new ListAppender<>();
+        listAppender.start();
+        context.getLogger(nodeImplLogger).addAppender(listAppender);
+    }
+
+    @After
+    public void loggingAppenderStop() {
+        listAppender.stop();
+        System.setProperty("oak.repository.property.logWarnStringSizeThreshold", "");
```

Comment:
You might want to use `TemporarySystemProperty` ... the rule would reset sys props to their initial state after test.

Similarly for logger, it might be easier (better??) to use `LogCustomizer` maybe

Reply:
Now using TemporarySystemProperty.
Going with logappender itself as I am also checking log message and logappender can compare message formats.

Reply:
Hmm... I wonder if it might be more useful to improve `LogCustomizer` to allow to compare message format. But, I'd leave that to you.
(Orthogonal to this conversation, looking at `LogCustomizer`, it seems it doesn't support to be used as a `Rule`. It would be interesting to expose a Rule which exposes a `LogCustomizerBuilder` type methods and takes care of calling `finished` when test ends)